src/Microsoft.Extensions.Configuration/ChainedConfigurationProvider.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Configuration
+{
+    public class ChainedConfigurationProvider : ConfigurationProvider
+    {
+        private readonly IConfiguration _configuration;
+
+        public ChainedConfigurationProvider(IConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+            _configuration = configuration;
+        }
+
+        public override void Load()
```

Comment:
Today all the providers I have looked at that represent an external source (i.e. not the in-memory one) eagerly copy all the data from the source into their own Data dictionary on `Load()`. So if the source changes Data becomes stale until you ask the configuration root to reload. `FileProvoderExtensions.ReloadOnChanged()` then can be use to automate the reloads.